addons/sale_management/models/sale_order.py

```diff
@@ -69,6 +69,12 @@ def _compute_validity_date(self):
             if validity_days > 0:
                 order.validity_date = fields.Date.context_today(order) + timedelta(validity_days)
 
+    @api.depends('sale_order_template_id')
+    def _compute_journal_id(self):
+        super()._compute_journal_id()
+        for order in self.filtered('sale_order_template_id'):
+            order.journal_id = order.sale_order_template_id.journal_id
```

Comment:
Avoid O(n²) complexity
```suggestion
    @api.depends('sale_order_template_id')
    def _compute_journal_id(self):
        super()._compute_journal_id()
        for order in self:
            if not order.sale_order_template_id:
                continue
            order.journal_id = order.sale_order_template_id.journal_id
```

Reply:
(was part of plan https://github.com/odoo/odoo/pull/133983) will solve in another PR